### ISSUE TITLE
java.lang.ArrayIndexOutOfBoundsException for strange CSV file

### ISSUE THREAD

**Issue by BitsReyesSantana** [wt-io-it/odoo-pycharm-plugin#55]
Plugin Version: 0.6.0
IntelliJ Version: PyCharm 2022.1.4
IntelliJ Build: PY-221.6008.17

<details><summary>com.intellij.diagnostic.PluginException: Failed to build index 'OdooIndex' for file file://G:/Trabajo/odoosh/Bits-process-automation/hr_recruitment_interview_survey/security/ir.model.access.csv (id = 590989) (file type = PLAIN_TEXT) [Plugin: at.wtioit.intellij.plugins.odoo]
</summary>

```
com.intellij.diagnostic.PluginException: Failed to build index 'OdooIndex' for file file://G:/Trabajo/odoosh/Bits-process-automation/hr_recruitment_interview_survey/security/ir.model.access.csv (id = 590989) (file type = PLAIN_TEXT) [Plugin: at.wtioit.intellij.plugins.odoo]
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:23)
	...
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:203)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	...
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: com.intellij.util.indexing.impl.MapReduceIndexMappingException: java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
	at com.intellij.util.indexing.impl.MapReduceIndex.mapInputAndPrepareUpdate(MapReduceIndex.java:250)
	at com.intellij.util.indexing.FileBasedIndexImpl.createSingleIndexValueApplier(FileBasedIndexImpl.java:1592)
	... 42 more
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
	at at.wtioit.intellij.plugins.odoo.OdooModelPsiElementMatcherUtil.getRecordsFromCsvFile(OdooModelPsiElementMatcherUtil.java:392)
	at at.wtioit.intellij.plugins.odoo.OdooModelPsiElementMatcherUtil.getRecordsFromCsvFile(OdooModelPsiElementMatcherUtil.java:382)
	at at.wtioit.intellij.plugins.odoo.records.index.OdooRecordFileIndex$OdooRecordFileIndexer.map(OdooRecordFileIndex.java:121)
	at at.wtioit.intellij.plugins.odoo.records.index.OdooRecordFileIndex$OdooRecordFileIndexer.map(OdooRecordFileIndex.java:101)
	at at.wtioit.intellij.plugins.odoo.index.OdooIndexExtension.map(OdooIndexExtension.java:26)
	at at.wtioit.intellij.plugins.odoo.index.OdooIndexer.lambda$map$0(OdooIndexer.java:19)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	...
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at at.wtioit.intellij.plugins.odoo.index.OdooIndexer.map(OdooIndexer.java:21)
	at at.wtioit.intellij.plugins.odoo.index.OdooIndexer.map(OdooIndexer.java:11)
	at com.intellij.util.indexing.impl.MapReduceIndex.mapByIndexer(MapReduceIndex.java:308)
	...
	at com.intellij.util.indexing.impl.MapReduceIndex.mapInputAndPrepareUpdate(MapReduceIndex.java:244)
	... 43 more
```
</details>

**Comment by bigbear3001** [comment]
@BitsReyesSantana thanks for reporting the issue, i have a fix in the queue, unfortunately along the way i broke something else with indexing and because i'm at Odoo Experience now, it might take until next week for a fix to be available.

**Comment by bigbear3001** [comment]
the issue is fixed in 0.6.1 (pending jetbrains approval) if you are eager to test it you can download the 0.6.1 jar from https://github.com/wt-io-it/odoo-pycharm-plugin/tree/v0.6.1

**Comment by bigbear3001** [comment]
0.6.1 got approved by jetbrains
